Fix elastic beanstalk bucket creation

ef4 opened the pull request:
This was calling a nonexistent method.

Perhaps everyone has been pre-creating their buckets, in which case they wouldn't hit this bug.

BanzaiMan commented on lib/dpl/provider/elastic_beanstalk.rb:
```diff
@@ -103,7 +103,7 @@ def bucket_exists?
 
       def create_bucket
-        s3.buckets.create(bucket_name)
+        s3.create_bucket({ bucket: bucket_name })
```
```ruby
s3.create_bucket(bucket: bucket_name)
```

I feel that it looks better without the brace here.

ef4 replied:
Haha, I agree. That's what happens when you spend years writing Javascript and haven't done Ruby in a long time. I will update the PR.

ef4 commented:
I updated the test suite to show the fix. It's all mocked of course, so it's not a high-fidelity test for compatibility with aws-sdk, but it's consistent with how the existing tests were already written.

In the test, I also changed:

> expect(provider).to receive(:s3).and_return(s3_mock).twice

to

> allow(provider).to receive(:s3).and_return(s3_mock)

Because I really don't think the number of calls to the `s3` method is relevant to the behavior under test. It's probably worth refactoring that out of all the other tests too and making it part of the setup.

BanzaiMan commented:
I'm closing this in favor of https://github.com/travis-ci/dpl/pull/638.
